src/domain/shared/utils/useJourneyCreation/useJourneyCreation.ts

```diff
@@ -0,0 +1,65 @@
+import { useCreateChallengeMutation, useCreateOpportunityMutation } from '../../../../hooks/generated/graphql';
+import { useApolloErrorHandler } from '../../../../core/apollo/hooks/useApolloErrorHandler';
+
+interface ChallengeCreationInput {
+  hubID: string;
+  displayName: string;
+  tagline: string;
+  background: string;
+  vision: string;
+  tags: string[];
+}
+
+interface OpportunityCreationInput {
+  challengeID: string;
+  displayName: string;
+  tagline: string;
+  vision: string;
+  tags: string[];
+}
+
+export const useJourneyCreation = () => {
+  const handleError = useApolloErrorHandler();
+
+  const [createChallengeLazy] = useCreateChallengeMutation({ onError: handleError });
```

Comment:
why lazy? Am I missing something? it's the same createChallenge as always no?

Reply:
there is nothing special about this, yes